be/src/exec/exec_node.cpp

```diff
@@ -308,10 +308,10 @@ Status ExecNode::collect_query_statistics(QueryStatistics* statistics) {
 }
 
 Status ExecNode::close(RuntimeState* state) {
-    if (_is_closed) {
+    if (_construct_state != INITIALIZED) {
         return Status::OK();
     }
```

Comment:
If it fails in the middle of `SubNode::init()` but already gains some resources, then `finish_init()` won't be invoked. Will the gained resources not be released?

Reply:
In theory, we should make the initialization procedure atomic, which could avoid all kinds of resource leak of double-free issues.

But in practice, the ExecNode only allocate some memory from the ObjectPool, which managing all resources. So there should not be any resource leak if initializing failed.